frontend/web/src/app/components/material/material-list/material-list.component.ts

```diff
@@ -0,0 +1,170 @@
+      field: 'id',
+      sortable: true,
+    }, {
+      title: 'Name',
+      field: 'name',
+      sortable: true,
+    }, {
+      title: 'Supplier name',
+      field: 'supplierName',
+      sortable: true,
+    }, {
+      title: 'Initial amount',
+      field: 'initialAmount',
+      sortable: true,
+    }, {
+      field: 'operate',
+      title: '<span class="glyphicon glyphicon-cog">',
+      align: 'center',
+      events: {
+        'click .edit'(e, value, row, index) {
+          _self.router.navigate([_self.router.url, row.id]);
+        },
+        'click .delete'(e, value, row, index) {
+          _self.delete(row.id);
+        },
+      },
+      formatter: this.operateFormatter,
+    }];
+
+    const data = [];
+    this.materials.forEach((material) => {
+      data.push({
+        id: material.id,
+        name: material.name,
+        supplierName: material.supplierName,
+        initialAmount: material.initialAmount,
+        materialGrades: material.materialGrades,
+        materialProperties: material.materialProperties,
+      });
+    });
+
+    (this.table as any).bootstrapTable('destroy');
+    (this.table as any).bootstrapTable(
+      {
+        data,
+        columns,
+        sortStable: true,
+        sortName: 'id',
+      },
+    );
+  }
+
+  private prepareTriggers() {
+    const _self = this;
+    (this.table as any).on('click-row.bs.table', (row, $element) => {
+      _self.selected.emit(_self.materials.filter((material) => { if (material.id === $element.id) { return material; } })[0]);
+    });
+  }
+
+  private amountFormatter(value, row, index) {
+    return value > 0 ? '' : {
+      css: { 'background-color': 'rgba(255, 0, 0, 0.4)' },
+    };
+  }
+
+  private operateFormatter(value, row, index) {
+    return [
+      '<button class="edit btn btn-xs btn-primary" href="javascript:void(0)" title="Edit">',
+      '<i class="glyphicon glyphicon-pencil"></i>',
+      '</button>  ',
+      '<button  class="delete btn btn-xs btn-danger" href="javascript:void(0)" title="Delete">',
+      '<i class="glyphicon glyphicon-trash"></i>',
+      '</button>',
+    ].join('');
+  }
+
+  private delete(id) {
+    this.toBeDeleted = id;
+    this.openModal('#deleteFormDismissBtn');
+  }
+
+  private dismissDelete() {
+    this.toBeDeleted = null;
+    this.modalRef.hide();
+  }
+
+  private confirmDelete() {
+    if (this.toBeDeleted) {
+      this.materialService.deleteMaterial(this.toBeDeleted).subscribe((res) => {
+        this.toBeDeleted = null;
+        this.modalRef.hide();
+        this.loadAndPopulate();
+      }, (error) => {
+        this.errorService.showAlert(error.verobose_message_header, error.verbose_message);
+      },
+      );
+    }
+  }
+
+  openModal(autoFocusIdWithHashtag: string) {
+    this.modalRef = this.modalService.show(this.modalDelete);
+    if (autoFocusIdWithHashtag != null && autoFocusIdWithHashtag !== '') {
+      const addInput: any = ($(autoFocusIdWithHashtag) as any);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)